src/cc/BPFTable.h

```diff
@@ -143,4 +250,41 @@ class BPFPerfBuffer : protected BPFTableBase<int, int> {
   std::unique_ptr<epoll_event[]> ep_events_;
 };
 
+class BPFProgTable : protected BPFTableBase<int, int> {
+private:
+  // this class is used to allow updating a table element using the
+  // table[index] = value syntax
+  class BPFProgTableWriter {
+    public:
+      BPFProgTableWriter(BPFProgTable& t, const int& i):
```

Comment:
Moreover, the update wrapper class are doing exact same thing across table types. Can we make it a templated class and reuse everywhere?